controllers/machinedeployment_sync.go

```diff
@@ -349,6 +350,14 @@ func (r *MachineDeploymentReconciler) scale(ctx context.Context, deployment *clu
 // syncDeploymentStatus checks if the status is up-to-date and sync it if necessary.
 func (r *MachineDeploymentReconciler) syncDeploymentStatus(allMSs []*clusterv1.MachineSet, newMS *clusterv1.MachineSet, d *clusterv1.MachineDeployment) error {
 	d.Status = calculateStatus(allMSs, newMS, d)
+
```

Comment:
may be add a comment here that `minReplicasNeeded` will be equal to `d.Spec.Replicas` when the strategy is not `RollingUpdateMachineDeploymentStrategyType`

Reply:
So the only other strategy I see implemented today is `OnDeleteMachineDeploymentStrategyType`. Would the comment become stale if we implement other strategies?

Reply:
```
	// Rolling update config params. Present only if
	// MachineDeploymentStrategyType = RollingUpdate.
	// +optional
	RollingUpdate *MachineRollingUpdateDeployment `json:"rollingUpdate,omitempty"`
```

As today maxUnavailable and MaxSurge are only applicable when `// MachineDeploymentStrategyType = RollingUpdate.`

I think reinforcing this would make it easier for new folks to ramp up and avoid confusion.

Reply:
Ok. Added the comment.